5 years on: Mitiq impact (#642)
unitaryfund/unitary.fund: src/content/blog/2024_mitiq_impact.md

---
title: The impact of Mitiq
author: nate stemen
day: 24
month: 12
year: 2024
tags: 
  - python
  - mitiq
  - community
---

> As we approach the 5 year anniversary since the Mitiq project has begun, we're starting to see how the project has played a role in shaping the ecosystem.

Reviewer: ```suggestion
As we approach the 5-year anniversary since the Mitiq project has begun, we're starting to see how the project has played a role in shaping the ecosystem.
```
This sentence makes it sound like we see the impact only now after 5 years, which sounds too negative imo.

Author: Good point. I'll rework this to say something more like "the five-year anniversary provides an opportunity to reflect how this project has shaped the ecosystem".

Preskill’s remarks underscore the enduring importance of error mitigation and highlight the need for an open-source, platform-agnostic tool like Mitiq to support the growing quantum ecosystem.
When Mitiq was first introduced in 2020, the landscape of QEM tools was nascent, with early efforts like pyquil’s measurement error mitigation and Qiskit Ignis paving the way.
Since then, the ecosystem has expanded, with Mitiq leading the charge by implementing nine QEM techniques and offering official support for 6 quantum SDKs (and un-official support for more).

Let’s take a closer look at how Mitiq has shaped the field and grown over time.

## Adoption and Community Growth

At the core of making QEM available to all, is building a platform-agnostic tool.
Whether you're writing your quantum program with Qiskit, Cirq, PyQuil, Qibo, Braket, ..., you can use Mitiq!
This has been a major driving force behind the growth and adoption of the project, and allows for users to write QEM code once with the flexibility of running on any (gate-based) hardware.
As the only SDK-agnostic QEM tool[^2], Mitiq’s adoption metrics provide a clear measure of its utility and impact.

> [^2]: To create such a tool in the ever-changing quantum software landscape that does not have an established IR, is, to say the least, a lot of work. In particular the **maintenance** overhead required to stay up to date with $n$ SDKs is quadratic.

Reviewer: nit: I don't like footnotes in blog post, but that may be my personal preference. They seem to me like a legacy of the past of paper printed stuff, which we should get rid of.

Reviewer: regarding the content of the footnote, why _quadratic_? We translate $n$ SDKs to Cirq, which is our internal representation, and that's linear. What am I missing?

Author: > footnotes

I'm going to keep the footnotes here, but I understand where you're coming from. How do you feel about "sidenotes"?

> quadratic

You're right, I was thinking about the generic case where we convert between every SDK. I'll remove the statement about it being quadratic, but keep the fact that it's a large maintenance overhead.

> - **Downloads:** Over 195,000 downloads (and counting!).

Reviewer: it should be clear from the graph below, but it's still better to specify this is _total_ since inception, and not /year

- **GitHub Engagement:** Mitiq has accumulated over 350 stars, 160 forks, and contributions from 76 developers worldwide.
- **Scientific Reach:** Researchers have cited Mitiq over 120 times since 2020 when the [whitepaper](https://arxiv.org/abs/2009.04417) was put online.

<div style="display: flex; justify-content: center; align-items: center;" class="side-by-side">
  <figure style="text-align: center;">
    <img src="/images/2024-mitiq/mitiq.png" width="600" alt="Bar chart of the Mitiq downloads per year from 2020 to 2024.">
    <figcaption>More than 50% growth in Mitiq downloads year over year!</figcaption>
  </figure>
</div>

The thriving Mitiq community is a testament to the importance of open-source collaboration.
From first-time contributors to seasoned quantum researchers, everyone has played a role in making Mitiq a vital tool for error mitigation.

## Driving Research and Innovation

> Mitiq has been cited in over 120 peer-reviewed research papers, enabling progress in areas such as condensed matter, quantum simulation, and benchmarking.

Reviewer: this sentence sounds a bit redundant so close to L42, which states the same statistics

Author: Good catch. Reworking.

> Some notable examples from the past 4 years include:

Reviewer: ```suggestion
Some notable examples from the past four years include:
```
However, I would stick with `five years`. At least that's what is in the title of this PR, and in the intro section.

Author: I'll simplify and go with five. I was thinking the whitepaper came out later than the project started, but I think that's an unnecessary details to reference.

> - [**Computational physics:**](https://arxiv.org/abs/2112.12654) Researchers used Mitiq's ZNE functionality to mitigate noise in quantum simulations of Heisenberg spin chains, enhancing fidelity and extending simulations beyond hardware coherence times.

Reviewer: As a reader of the rendered version, it's not clear to me that those are links leading to papers. Maybe you want to add the actual title of the paper as text of the link.

Author: Good point. I like your suggestion.

- **Presentations:** Since Mitiq's inception, Unitary Fund staff, alongside Mitiq contributors have delivered over 50 talks to the scientific and programming communities. This year we presented at the APS March meeting, IEEE QCE, and unitaryCON.
- **Mitiq workshop:** This year we hosted a workshop dedicated solely to teaching people how to use, contribute to, and experiment with Mitiq in conjunction with the [QNumerics summer school](https://qnumerics.org/).
- **Project mentorship:** We've continued to grow our connections with students by mentoring QEM projects through [QRISE](https://www.quantumcoalition.io/qrise), the [University of Washington's capstone course](https://unitary.fund/posts/2024_capstone_uw/), and a new project coming soon in conjuction with the University of Amsterdam.

## Looking Ahead

The space is growing... a lot!
It's hard to get a sense of project usage from open-source software stats, but when many similar projects are seeing upticks in the number of downloads, it is a stronger signal that tools related to error mitigation/suppression/characterization are in high demand.

> Below you can see the download number from two other open-source projects, Qermit (error mitigation) and pyGSTi (tomography).

Reviewer: I like this comparison!

In addition to noise characterization, we've seen two other needs from our community.

1. On the software side, tooling is needed for QEM technique comparison and combination, and
2. on the research side, the industry needs a better understanding of how QEM will interplay with QEC.

We hope to make progress on both of these in 2025, and we won't be able to without your help!
We need fresh ideas, people writing code, doing research, giving talks, etc.
If this sounds like something you're interested in, I encourage you to get involved.

## Join the Mitiq Community

Whether you’re a researcher, educator, or software engineer, or quantum tinkerer, there’s a place for you in the Mitiq community.

- Check out the [`mitiq`](https://github.com/unitaryfund/mitiq) GitHub repository and contribute by opening issues, asking questions, or making a pull request. If you have a QEM technique you'd like to use, let us know! Feature requests are always appreciated, and encouraged.
- Join our [Discord community](http://discord.unitary.fund) to connect with me and all the other wonderful Mitiq developers and quantum enthusiasts.

> - Sign up to recieve the [Mitiq Newsletter](https://forms.gle/6UcUjSawHyweXhQV7) to stay up to date with the latest happenings.

Reviewer: I am going to add this call to action at the bottom of the newsletter as well. Someone may forward the newsletter to a friend/colleague and, in turn, they may want to sign up.

> - Sign up to recieve the [Mitiq Newsletter](https://forms.gle/6UcUjSawHyweXhQV7) to stay up to date with the latest happenings.

Reviewer: ```suggestion
- Sign up to receive the [Mitiq Newsletter](https://forms.gle/6UcUjSawHyweXhQV7) to stay up to date with the latest happenings.
```

Have we acheived this goal?
Making QEM easily available to **everyone** is an ambitious target, but as the adoption metrics below demonstrate, Mitiq has significantly lowered the barrier to using Quantum Error Mitigation (QEM) protocols.

As the field of Quantum Error Mitigation matured, Mitiq has gained recognition as an optional, but important component of the quantum computing stack.
While some skeptics argue that Quantum Error Correction (QEC) will eventually render QEM obsolete, others, like John Preskill, believe that the two approaches will complement one another even in the long term.
In his [talk at Q2B 2024](https://preskill.caltech.edu/talks/Preskill-Q2B-2024.pdf), Preskill stated:

> <p style="font-size: 24px;">Error mitigation will continue to be useful in the Megaquop era and beyond.</p>

Preskill's remarks underscore the enduring importance of error mitigation and highlight the need for an open-source, platform-agnostic tool like Mitiq to support the growing quantum ecosystem.
When Mitiq was first introduced in 2020, the landscape of QEM tools was nascent, with early efforts like pyquil's measurement error mitigation and Qiskit Ignis paving the way.

> Since then, the ecosystem has expanded, with Mitiq leading the charge by implementing nine QEM techniques and offering official support for 6 quantum SDKs (and un-official support for more).

Reviewer: ```suggestion
Since then, the ecosystem has expanded, with Mitiq leading the charge by implementing nine QEM techniques and offering official support for 6 quantum SDKs (and unofficial support for more).
```

Let's take a closer look at how Mitiq has shaped the field and grown over time.

## Adoption and Community Growth

At the core of making QEM available to all, is building a platform-agnostic tool.
Whether you're writing your quantum program with Qiskit, Cirq, PyQuil, Qibo, Braket, ..., you can use Mitiq!
This has been a major driving force behind the growth and adoption of the project, and allows for users to write QEM code once with the flexibility of running on any (gate-based) hardware.
As the only SDK-agnostic QEM tool[^2], Mitiq's adoption metrics provide a clear measure of its utility and impact.

[^2]: To create such a tool in the ever-changing quantum software landscape that does not have an established IR, is, to say the least, a lot of work. In particular the **maintenance** required to stay up to date with $n$ SDKs and conversions between.

- **Downloads:** Over 195,000 total downloads (and counting!).

> - **GitHub Engagement:** Mitiq has accumulated over 350 stars, 160 forks, and contributions from 76 developers worldwide.

Reviewer: 360+ stars as of now :D

---
title: Reflecting on the Impact of Mitiq and the Future of Quantum Error Mitigation
author: nate stemen
day: 24
month: 12
year: 2024
tags: 
  - python
  - mitiq
  - community
---

> **Five years since Mitiq's inception and much progress has been made, but still with many opportunities for new and exciting work.**

Reviewer 2: ```suggestion
**Much progress has been made since Mitiq's inception five years ago, but still with many opportunities for new and exciting work.**
```
Maybe it's me but the way this is written feels awkward.

Have we achieved this goal?
Making QEM easily available to **everyone** is an ambitious target, but as the adoption metrics below demonstrate, Mitiq has significantly lowered the barrier to using Quantum Error Mitigation (QEM) protocols.

As the field of Quantum Error Mitigation has matured, Mitiq has gained recognition as an important component of the quantum computing stack.
While some argue that Quantum Error Correction (QEC) could eventually render QEM obsolete, the two techniques *can* work in tandem.
For example, see [Unitary Fund's work](https://unitary.fund/posts/2023_mitiq_stim_workflow/) demonstrating an approach using Mitiq together with a leading QEC tool, Stim.
In a [talk at Q2B 2024](https://preskill.caltech.edu/talks/Preskill-Q2B-2024.pdf) John Preskill emphasized that the two approaches will complement one another, even in the long term.

> <p style="font-size: 24px;">Error mitigation will continue to be useful in the Megaquop era and beyond.</p>

Preskill's statement underscores the importance of error mitigation and highlight the need for an open-source, platform-agnostic tool like Mitiq to support the growing quantum ecosystem.

> When Mitiq was first introduced in 2020, the landscape of QEM tools was nascent, with early efforts like pyQuil's measurement error mitigation and Qiskit Ignis paving the way.

Reviewer 2: ```suggestion
When Mitiq was first introduced in 2020, the landscape of QEM tools was nascent, with early efforts like [pyQuil's measurement error mitigation](add link here) and [Qiskit Ignis](https://github.com/qiskit-community/qiskit-ignis) paving the way.
```

I tried to search for a link to pyquil's QEM tool.

  <figure style="text-align: center;">
    <img src="/images/2024-mitiq/mitiq.png" width="600" alt="Bar chart of the Mitiq downloads per year from 2020 to 2024.">
    <figcaption>More than 50% growth in Mitiq downloads year over year!</figcaption>
  </figure>
</div>

The Quantum Open-Source Software Survey this year indicated that 8.8% of respondents were using Mitiq, with an additional 4.3% planning to use it in the next year.
This positions Mitiq as the [10th most used tool](https://unitaryfund.github.io/survey-2024/#softwares-for-applications-and-tools-used-currently-or-in-the-future) in the quantum software space&mdash;or **5th** if all Qiskit tools are grouped together.

In addition to the tools high adoption, the thriving Mitiq community is a testament to the importance of open-source collaboration.
From first-time contributors to experienced quantum researchers, everyone has played a role in elevating Mitiq to the tool it is today.

## Driving Research and Innovation

Mitiq has been an instrumental tool in advancing research across diverse areas of quantum computing, including benchmarking, quantum simulation, and QEM research.

> Its easy-to-use API, combined with extensive documentation, has empowered researchers from fields like condensed matter and many-body physics to explore techniques that push the boundary of what's possible on today's quantum hardware.

Reviewer 2: ```suggestion
Its easy-to-use [API](https://github.com/qiskit-community/qiskit-ignis), combined with [extensive documentation](https://mitiq.readthedocs.io/en/stable/guide/guide.html), has empowered researchers from fields like condensed matter and many-body physics to explore techniques that push the boundary of what's possible on today's quantum hardware.
```

Notable examples include:

- **Computational physics:** Researchers used Mitiq's Zero-Noise Extrapolation (ZNE) functionality to mitigate noise in quantum simulations of Heisenberg spin chains, enhancing fidelity and extending simulations beyond hardware coherence times. [[*Quantum dynamics simulations beyond the coherence time on NISQ hardware by variational Trotter compression*](https://arxiv.org/abs/2112.12654)]

> - **Statistical physics:** Researchers used Mitiq to demonstrate a reduction in noise during quantum simulations of Bethe eigenstates. Their experiments, the authors (of [*Algebraic Bethe Circuits*](https://quantum-journal.org/papers/q-2022-09-08-796/)) say,

Reviewer 2: ```suggestion
- **Statistical physics:** Researchers used Mitiq to demonstrate a reduction in noise during quantum simulations of Bethe eigenstates. Their experiments, to quote the authors (of [*Algebraic Bethe Circuits*](https://quantum-journal.org/papers/q-2022-09-08-796/)),
```

## Education and Outreach

Mitiq has not only advanced research but also contributed to education and training in quantum computing.
In January 2024 I had the pleasure to interview more than 50 people across the quantum computing industry to discuss the challenges they faced in using quantum computers.
This was part of our homework after receiving an [NSF POSE award](https://unitary.fund/posts/2023_mitiq_nsf_pose/) in order to better understand Mitiq's positioning in the larger software/hardware ecosystem, and to identify gaps in tooling.
One thing that was touched on by nearly all interviewees who had contributed to Mitiq at some point was how easy it was to get started.
We pride ourselves on maintaining a well-documented project (even for developer workflows) and offering informal mentorship opportunities.
Our industry needs talent, and Mitiq has played a key role in onboarding individuals with essential skills—such as software development, research, and technical writing—into the field.
Some of the key ways that we work to encourage this include:

> - **Documentation:** Mitiq has over 100 pages of documentation for our QEM techniques, showing both the theory and practice of applying QEM on real devices. This includes 35 examples to base your next project on as well as a [short video explainer](https://www.youtube.com/watch?v=47GWi4h7TWM) of our latest technique.

Reviewer 2: ```suggestion
- **Documentation:** Mitiq has over 100 pages of documentation for our QEM techniques, showing both the theory and practice of applying QEM on real devices. This includes [35 examples](https://mitiq.readthedocs.io/en/stable/examples/examples.html) to base your next project on as well as a [short video explainer](https://www.youtube.com/watch?v=47GWi4h7TWM) of our latest technique.
```

- **Project mentorship:** We've continued to grow our connections with students by mentoring QEM projects through [QRISE](https://www.quantumcoalition.io/qrise), the [University of Washington's capstone course](https://unitary.fund/posts/2024_capstone_uw/), and a new project coming soon in conjuction with the University of Amsterdam.

An encouraging indicator of our success in maintaining an easy-to-contribute-to project is Mitiq’s relatively high ratio of contributors to users. With nearly 200,000 downloads and 77 contributors, Mitiq stands out compared to other projects: QuTiP, for instance, has 167 contributors for 2 million downloads, and Qiskit has 592 contributors for 8.5 million downloads.

## Looking Ahead

The space is growing... a lot!
It's hard to get a sense of project usage from open-source software stats, but when many similar projects are seeing upticks in the number of downloads, it is a stronger signal that tools related to error mitigation/suppression/characterization are in high demand.

> Below you can see the download number from two other open-source projects, Qermit (error mitigation) and pyGSTi (tomography).

Reviewer 2: ```suggestion
Below you can see the download number from two other open-source projects, [Qermit (error mitigation)](https://github.com/CQCL/Qermit) and [pyGSTi (tomography)](https://github.com/sandialabs/pyGSTi).
```

<style>
  .side-by-side {
    display: flex;
    justify-content: center; /* Center align the images */
    gap: 10px; /* Add spacing between images */
  }
  .side-by-side img {
    max-width: 100%;
    height: auto;
    border-radius: 8px; /* Optional rounded corners */
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); /* Optional shadow */
  }
  .side-by-side div {
    text-align: center; /* Center captions under images */
    width: 45%; /* Control image width */
  }
</style>

<div class="side-by-side">
  <div>
    <img src="/images/2024-mitiq/qermit.png" alt="Bar chart of the Qermit downloads per year from 2021 to 2024.">
    <p>Qermit saw a 100% increase in downloads YoY!</p>
  </div>
  <div>
    <img src="/images/2024-mitiq/pygsti.png" alt="Bar chart of the pyGSTi downloads per year from 2016 to 2024.">
    <p>pyGSTi amassed over 120k downloads in a calendar year!</p>
  </div>
</div>

Two takeaways from these plots are

1. QEM as a whole is growing, not just Mitiq, and
2. lots of people are running tomography experiments.

> These two facts, taken together with the lack of tooling[^1] to take noise characterization data to inform QEM showcase a gap in our ecosystem.

Reviewer 2: ```suggestion
These two facts, combined with the dearth of tools[^1] that could provide noise characterization data to inform QEM showcase a gap in our ecosystem.
```
